bigquery_etl/shredder/delete.py

```diff
@@ -223,30 +224,76 @@ def create_job(client):
 
 def get_partition_expr(table):
     """Get the SQL expression to use for a table's partitioning field."""
-    for field in table.schema:
-        if field.name != table.time_partitioning.field:
-            continue
-        if field.field_type == "TIMESTAMP":
-            return f"DATE({field.name})"
-        return field.name
+    if table.range_partitioning:
+        return table.range_partitioning.field
+    if table.time_partitioning:
+        return next(
+            f"DATE({field.name})" if field.field_type == "TIMESTAMP" else field.name
+            for field in table.schema
+            if field.name == table.time_partitioning.field
+        )
```

Comment:
```suggestion
        return f"CAST({field.name} AS DATE)".format(field=table.time_partitioning.field)
```

I believe this form is tolerant of both TIMESTAMP and DATE without having to determine the underlying type, but we should check that it is still usable for partition selection.

Reply:
> I believe this form is tolerant of both TIMESTAMP and DATE without having to determine the underlying type, but we should check that it is still usable for partition selection.

ooh, nice. i'll check and use if so.

Reply:
🎉 cast works!